component.js

```diff
@@ -255,9 +259,21 @@ function factory (options) {
     }
 
     if (!hasShouldComponentUpdate(mixins)) {
-      mixins = [{
+      mixins.unshift({
         shouldComponentUpdate: _shouldComponentUpdate
-      }].concat(mixins);
+      });
+    }
+
+    if (!hasComponentWillReceiveProps(mixins)) {
```

Comment:
This is related to https://github.com/omniscientjs/omniscient/pull/68#discussion_r25449500 . Life cycle methods are guaranteed to be invoked, so there won't be any need for this. Good thought, though :+1: